File: 프로그래머스/lv1/12901. 2016년/2016년.kt
class Solution {
    fun solution(a: Int, b: Int): String {
        val day = monthToDate(a) + b
        return dateToDay(day%7)
    }

    fun monthToDate(month: Int): Int {
        return when(month) {
            1 -> 0
            2 -> 31
            3 -> 60
            4 -> 91
            5 -> 121
            6 -> 152
            7 -> 182
            8 -> 213
            9 -> 244
            10 -> 274
            11 -> 305
            else -> 335
        }
    }

    fun dateToDay(date: Int): String {
        return when(date) {
            0 -> "THU"
            1 -> "FRI"
            2 -> "SAT"
            3 -> "SUN"
            4 -> "MON"
            5 -> "TUE"
            else -> "WED"
        }
    }
}

File: 프로그래머스/1/12901. 2016년/2016년.swift
func solution(_ a:Int, _ b:Int) -> String {
    let week = ["THU", "FRI", "SAT", "SUN", "MON", "TUE", "WED"]
    
    return week[getDay(a, b)]
}

func getDay(_ a: Int, _ b: Int) -> Int {
    let calendar = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
    var day = 0
    
    for i in 0...a-1 {
        day += (i == a-1) ? b : calendar[i]
    }
    
    return day % 7
}